js/app/test/clear_components.spec.ts

```diff
@@ -0,0 +1,28 @@
+import { test, expect } from "@gradio/tootils";
+
+test("gr.ClearButton clears every component's value", async ({ page }) => {
+	await Promise.all([
+		page.waitForResponse("**/run/predict"),
+		page.click("text=Get Values")
+	]);
+	await expect(page.getByLabel("Are all cleared?")).toHaveValue("False");
+	await page.click("text=Clear");
+	await Promise.all([
+		page.waitForResponse("**/run/predict"),
+		page.click("text=Get Values")
+	]);
+	await expect(page.getByLabel("Are all cleared?")).toHaveValue("True");
+});
+
+test("Components can be hidden and removed from page", async ({ page }) => {
+	[...Array(5).keys()].forEach(async (val) => {
```

Comment:
Having some difficulty testing all components. I think because not all of them fit in the page, some are not visible when they should be. But I think this is still good enough.